src/org/jitsi/impl/neomedia/MediaStreamImpl.java

```diff
@@ -3678,10 +3688,47 @@ public boolean isEndOfFrame(byte[] buf, int off, int len)
     }
 
     /**
-    * {@inheritDoc}
-    */
+     * {@inheritDoc}
+     * </p>
+     * This is absolutely terrible, but we need a RawPacket and the method is
+     * used from RTPTranslator, which doesn't work with RawPacket.
+     */
     public boolean isKeyFrame(byte[] buf, int off, int len)
     {
+        return isKeyFrame(new RawPacket(buf, off, len));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public boolean isKeyFrame(RawPacket pkt)
+    {
+        byte[] buf = pkt.getBuffer();
+        int off = pkt.getOffset();
+        int len = pkt.getLength();
+
+        if (frameMarkingsExtensionId != -1)
+        {
+            RawPacket.HeaderExtensions hes = pkt.getHeaderExtensions();
```

Comment:
in order for the above recommendation to work, the getHeaderExtensions() can be made static.